Mathlib/Topology/ProperMap.lean

```diff
@@ -136,6 +137,60 @@ lemma IsProperMap.ultrafilter_le_nhds_of_tendsto (h : IsProperMap f) ⦃𝒰 : U
     (hy : Tendsto f 𝒰 (𝓝 y)) : ∃ x, f x = y ∧ 𝒰 ≤ 𝓝 x :=
   (isProperMap_iff_ultrafilter.mp h).2 hy
 
+/-- The composition of two proper maps is proper. -/
+lemma IsProperMap.comp (hf : IsProperMap f) (hg : IsProperMap g) :
+    IsProperMap (g ∘ f) := by
+  refine ⟨by continuity, fun ℱ z h ↦ ?_⟩
+  rw [MapClusterPt, ← map_map] at h
+  rcases hg.clusterPt_of_mapClusterPt h with ⟨y, hy1, hy2⟩
+  rcases hf.clusterPt_of_mapClusterPt hy2 with ⟨x, hx1, hx2⟩
+  use x
+  rw [← hy1, ← hx1]
+  exact ⟨rfl, by assumption⟩
+
+/-- If the composition of two continuous functions `g ∘ f` is proper and `f` is surjective,
+then `g` is proper. -/
+lemma isProperMap_of_comp_of_surj (hf : Continuous f)
+    (hg : Continuous g) (hgf : IsProperMap (g ∘ f)) (f_surj : f.Surjective) : IsProperMap g := by
+  refine ⟨hg, fun ℱ z h ↦ ?_⟩
+  rcases hgf with ⟨_, h'⟩
+  rw [← inf_top_eq ℱ, ← principal_univ, ← f_surj.range_eq, ← ℱ.map_comap f, MapClusterPt,
+    map_map, ← MapClusterPt] at h
+  rcases h' h with ⟨x, hx1, hx2⟩
+  refine ⟨f x, hx1, neBot_of_comap (m := f) (NeBot.mono hx2 ?_)⟩
+  calc
+    𝓝 x ⊓ comap f ℱ
+      ≤ comap f (𝓝 (f x)) ⊓ comap f ℱ := inf_le_inf_right _ (hf.tendsto x).le_comap
+    _ = comap f (𝓝 (f x) ⊓ ℱ) := comap_inf.symm
```

Comment:
You can simplify this a bit by using again the equality `ℱ = map f (comap f ℱ)`, together with [ClusterPt.map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Basic.html#ClusterPt.map):
```suggestion
  rw [← ℱ.map_comap_of_surjective f_surj]
  exact ⟨f x, hx1, hx2.map hf.continuousAt tendsto_map⟩
```